Add use of MutationObserver to fix issues with new Google Calendar UI

apphancer opened the pull request:
This commit introduces the use of MutationObserver in order to address issues with Google's new Calendar UI.
MutationObserver observes for changes in the DOM and with this it's possible to apply the disable scroll function each time there is a relevant event change (e.g. changing month, week, calendar format, create a new event).

apphancer committed:
Add use of MutationObserver to fix issues with new Google Calendar UI

DavGoodvibes commented:
How to try version #10 ? The one I tried was released by hamen as zip file : 
[google-calendar-scroll-disabler#9.zip](https://github.com/alter-ego/google-calendar-scroll-disabler/files/1404192/google-calendar-scroll-disabler.zip)

apphancer commented:
Download my revised code from here https://github.com/apphancer/google-calendar-scroll-disabler
And install the extension locally

DavGoodvibes commented:
apphancer : v0.3 is working for me. It look that you have your new version ;-)
Congratulations & thanks for your efforts.

apphancer committed:
Fix call to observe_if_calendar_available() from window.setTimeout

apphancer committed:
Remove dependencies to jQuery and jQuery library

apphancer commented:
Thanks for your feedback @DavGoodvibes 

I have made a further commit where I have removed the jQuery library and changed 2 lines in content.js file lines to use VanillaJS

x1c0 commented:
Good job @apphancer! I was also trying to do this myself, first i also remove the use of jQquery but then I was struggling with the main grid changing and I could not find anything in the DOM that was changing.. the div that contains the main calendar is the same but after changing the month the listener was not firing.. 
Very clever approach with MutationObserver! I was not even aware that this existed 🙈 and i'm still not 100% sure how you do it with that `'data-viewfamily'` 😂 

I also fully tested this PR and it works perfectly👌 

Thank you!

x1c0 commented on code/content.js:
```diff
@@ -1,17 +1,38 @@
+var disable_scroll = function () {
+    for (var live_selector of document.querySelectorAll(calendar_grid_selector)) {
+        live_selector.addEventListener('mousewheel', function (e) {
             if (e.target.id == 'el') return;
```
Is this if condition still needed? I could not find any purpose for this, maybe this was needed on the previous Calendar version?

danielosborn commented:
This branch/PR worked great for me (installed locally). Thank you for you work on this! Such an annoying "feature", I don't know why Google insists on keeping it or not having a way to turn it off.
